afs/templates/views/components/reports/scenes/existence.htm

```diff
@@ -6,72 +6,75 @@
 <div data-bind="foreach: {data: events, as: 'eventName', noChildContext: true}" class="afs-report-section">
     <h2 data-bind="text: eventName + ' Event'" class="afs-report-section-title"></h2>
     <span data-bind="if: cards[eventName] && cards[eventName].card && !cards[eventName].card.tiles().length,">
-        <a data-bind="{click:function(){addNewTile(cards[eventName].card)}}" class="afs-report-a">
+        <a data-bind="{click:function(){$parent.addNewTile(cards[eventName].card)}}" class="afs-report-a">
             <i class="fa fa-mail-reply"></i> {% trans "Add Event" %}
         </a>
     </span>
     <span data-bind="if: cards[eventName] && cards[eventName].card && cards[eventName].card.tiles().length">
-    <a data-bind="{click:function(){addNewTile(cards[eventName].card)}}" class="afs-report-a">
+    <a data-bind="{click:function(){$parent.addNewTile(cards[eventName].card)}}" class="afs-report-a">
         <i class="fa fa-mail-reply"></i> {% trans "Edit Event" %}
     </a>
-    </span><span data-bind="text: console.log(eventData)"></span>
-    <!-- ko if: !eventData[eventName] || !eventData[eventName].tileid -->
+    <!-- ko if: eventDataArray[eventName].length === 0 || !eventDataArray[eventName].some(x => x !== undefined) -->
     <div class="afs-nodata-note">No events</div>
     <!-- /ko -->
+
     <!-- Show this div if there are event data -->
     <!-- Event Block (one for each event) -->
-    <!-- ko if: eventData[eventName] && eventData[eventName].tileid --> 
+    <!-- ko if: eventDataArray[eventName].length > 0 && eventDataArray[eventName].some(x => x !== undefined) -->
+    
+    <!-- ko foreach: { data: eventDataArray[eventName], as: "eventData" } -->
```

Comment:
It's worth noting that setting noChildContext to true here is probably the best solution since you're already using an "as" reference - then you wouldn't need to go through and modify the parent context for all of these child elements.  Note above that where you're adding it - you shouldn't need to add it, but it works because the $parent is whatever's calling existence, which will also happen to inherit from the report.js helper.  The solution works, but I've found nested contexts painful to deal with.  Your choice on whether to change or not.  Let me know if you'd prefer keep as is and I'll resolve this comment.